pyproject.toml

```diff
@@ -26,24 +26,24 @@ classifiers=[
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-"numpy",
-"accelerate>=0.20.3",
-"transformers",
-"torch",
-"sentencepiece",
-"tokenizers>=0.13.3",
-"tqdm",
-"trl",
-"peft>=0.8.0",
-"datasets>=2.15.0",
-"fire",
-"simpleeval",
+"numpy>=1.26.4,<2.0",
+"accelerate>=0.20.3,<1.0",
+"transformers>=4.34.1,<5.0,!=4.38.2",
+"torch>=2.2.0,<3.0",
+"sentencepiece>=0.1.99,<1.0",
+"tokenizers>=0.13.3,<1.0",
+"tqdm>=4.66.2,<5.0",
+"trl>=0.7.10,<1.0",
+"peft>=0.8.0,<1.0",
+"datasets>=2.15.0,<3.0",
+"fire>=0.5.0,<1.0",
+"simpleeval>=0.9.13,<1.0",
 ]
 
 [project.optional-dependencies]
-dev = ["wheel", "packaging", "ninja", "scikit-learn>=1.0, <2.0"]
-flash-attn = ["flash-attn"]
-aim = ["aim==3.19.0"]
+dev = ["wheel>=0.42.0,<1.0", "packaging>=23.2,<24", "ninja>=1.11.1.1,<2.0", "scikit-learn>=1.0, <2.0"]
+flash-attn = ["flash-attn>=2.5.3,<3.0"]
+aim = ["aim>=3.19.0,<4.0"]
```

Comment:
Note that from [PR #93 comment](https://github.com/foundation-model-stack/fms-hf-tuning/pull/93#issuecomment-2100720341)
> [aim] was initially capped to specific version as newer major version of Aim contained some stability issues. Now I feel we can keep it directly to 3.X.X